xgcm/comodo.py

```diff
@@ -119,12 +124,21 @@ def _maybe_fix_type(attr):
                     "length %g (axis_len=%g)" % (name, clen, axis_len)
                 )
         else:
-            raise ValueError(
-                "Coordinate %s has invalid or missing "
-                "`c_grid_axis_shift` attribute `%s`" % (name, repr(shift))
-            )
-    return axis_coords
+            if shift not in valid_axis_shifts:
+                # string representing valid axis shifts
+                valids = str(valid_axis_shifts)[1:-1]
 
+                raise ValueError(
+                    "Coordinate %s has invalid "
+                    "`c_grid_axis_shift` attribute `%s`. "
+                    "`c_grid_axis_shift` must be one of: %s" % (name, repr(shift), valids)
+                )
+            else:
+                raise ValueError(
```

Comment:
I think the [comodo spec](https://web.archive.org/web/20160417032300/http://pycomodo.forge.imag.fr/norm.html#attribute-c-grid-axis-shift) inteprets a 'missing' `c_grid_axis_shift` as `0` or `centered`. So this might be too strict of a requirement?

Reply:
I've added this to list. The code doesn't actually prevent 0 being used, but better to have this in the error message, so I've included it in the update.